rust/agama-software/src/zypp_server.rs

```diff
@@ -375,6 +373,39 @@ impl ZyppServer {
         Ok(())
     }
 
+    fn select_resolvable(
+        &self,
+        zypp: &zypp_agama::Zypp,
+        name: &str,
+        r#type: ResolvableType,
+        reason: zypp_agama::ResolvableSelected,
+        optional: bool,
+    ) -> Vec<Issue> {
+        let mut issues = vec![];
+        let result = zypp.select_resolvable(name, r#type.into(), reason);
+
+        if let Err(error) = result {
+            if optional {
```

Comment:
well, it can cloak real errors. Ideally we should for optional at first ask if resolvable is available ( not available now in agama-zypp ) and skip if not.

so ideally something like
```
if optional && !zypp.resolvable_available(name, r#typ.into()) {
  return issues;
}
```